iris_doc/api_tagger.py

```diff
@@ -254,13 +271,17 @@ def _findClassScopeEndIndex(self, lines: List[str], scopeStartIndex: int) -> int
         scopeStack.append(scopeStartIndex)
 
         i = scopeStartIndex + 1
+        has_match_function_start = False
 
-        while(i < len(lines)):
+        while (i < len(lines)):
             line = lines[i].strip()
-            if self.__syntaxMatcher.matchClassScopeEnd(line) or self.__syntaxMatcher.matchFunctioinScopeEnd(line):
+            if self.__syntaxMatcher.matchClassScopeEnd(line) or \
+                    (has_match_function_start and self.__syntaxMatcher.matchFunctioinScopeEnd(line)):
                 scopeStack.pop()
+                has_match_function_start = False
             elif self.__syntaxMatcher.matchFunctioinScopeStart(line):
                 scopeStack.append(i)
+                has_match_function_start = True
```

Comment:
I just tried the following in my branch, removing the ObjCLineScanner check and it passes the tests in all languages:
```py
        while(i < len(lines)):
            line = lines[i].strip()
            if self.__syntaxMatcher.matchFunctionScopeStart(line) or self.__syntaxMatcher.matchClassScopeStart(line):
                scopeStack.append(i)
            if self.__syntaxMatcher.matchClassScopeEnd(line) or (not self.__syntaxMatcher.matchMemberVariable(line) and self.__syntaxMatcher.matchFunctionScopeEnd(line)):
                scopeStack.pop()
            if len(scopeStack) == 0:
                return i
```

Functions and properties in objc header files end the same way, with a ";"

Reply:
@maxxfrazer Updated.